fiat-amd64/gentest.py

```diff
@@ -28,32 +32,58 @@ def removeprefix(s, prefix):
   p384='2^384 - 2^128 - 2^96 + 2^32 - 1',
   p384_scalar='2^384 - 1388124618062372383947042015309946732620727252194336364173',
   p434='2^216 * 3^137 - 1',
-  secp256k1='2^256 - 2^32 - 977',
+  secp256k1_montgomery='2^256 - 2^32 - 977',
   secp256k1_scalar='2^256 - 432420386565659656852420866394968145599')
 
+saturatedsolinasprimes = dict(
+  curve25519_solinas=('2^255 - 19'))
+
+dettmanprimes = dict( # last limb width, limbs, prime
+  secp256k1_dettman=('48', '5', '2^256 - 4294968273'))
+
 output_makefile = ('--makefile' in sys.argv[1:])
-asm_files = tuple(i for i in sys.argv[1:] if i not in ('--makefile',))
+directories = tuple(i for i in sys.argv[1:] if i not in ('--makefile',))
 
 asm_op_names = OrderedDict()
-for fname in asm_files:
-    op, name = removesuffix(fname, '.asm').replace('_solinas','').split('_')[-2:]
-    asm_op_names.setdefault((name, op), []).append(fname)
+
+regex = re.compile(r'fiat_(?P<name>[^_]+(_((solinas)|(montgomery)|(dettman)))?)_(?P<op>(carry_)?((square)|(mul)))')
+for dirname in directories:
+    m = regex.match(os.path.basename(dirname)).groupdict(None)
+    if m is None:
+        continue
+    for fname in os.listdir(dirname):
+        asm_op_names.setdefault((m.get('name'), m.get('op')), []).append(os.path.join(dirname,fname))
```

Comment:
Minor: I would write:
```suggestion
    if m:
        for fname in os.listdir(dirname):
            asm_op_names.setdefault((m.get('name'), m.get('op')), []).append(os.path.join(dirname, fname))
```

Reply:
I also found it rather verbose to use `(m.get('name'), m.get('op'))` as the key. Could that not just be `m` in this case?

Reply:
`m` is a dictionary, so you don't want to use it as a key.  You could plausibly do something like `tuple(sorted(m.items()))`, but that seems needlessly arcane.  You can shorten it by doing `(m['name'], m['op'])`, though, I think.  Or you can use another line, do `key = (m['name'], m['op'])` and then do `.setdefault(key, []).append(...)`